File: GiftIdeaTracker/src/GiftIdeaTracker.WebApp/projects/gift-idea-tracker/src/app/models/gift-idea.ts
import { Occasion } from './occasion';
import { Recipient } from './recipient';

export interface GiftIdea {
  giftIdeaId: string;
  userId: string;
  recipientId: string | null;
  recipient?: Recipient;
  name: string;
  occasion: Occasion;
  estimatedPrice: number | null;
  isPurchased: boolean;
  createdAt: string;
}

export interface CreateGiftIdeaRequest {
  recipientId?: string;
  name: string;
  occasion: Occasion;
  estimatedPrice?: number;
}

export interface UpdateGiftIdeaRequest {
  giftIdeaId: string;
  recipientId?: string;
  name: string;
  occasion: Occasion;
  estimatedPrice?: number;
}

File: GiftIdeaTracker/src/GiftIdeaTracker.Core/Model/GiftIdeaAggregate/GiftIdea.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace GiftIdeaTracker.Core;

public class GiftIdea
{
    public Guid GiftIdeaId { get; set; }
    public Guid UserId { get; set; }
    public Guid? RecipientId { get; set; }
    public Recipient? Recipient { get; set; }
    public string Name { get; set; } = string.Empty;
    public Occasion Occasion { get; set; }
    public decimal? EstimatedPrice { get; set; }
    public bool IsPurchased { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
